### PR TITLE
Ref/feature/undo menu bar

### PR DESCRIPTION
**Pull request by MichiGandi**
Add `Undo` and `Redo` to the menu bar.

To achieve this, multiple changes were made:
- Merge the position index of the menu bar system into the path to be able to give each path element its index.
- Add sorting to the context menu system. This allows the implementation of the following step.
- replace position index with sorting order. The position index was relative to the number of already existing items, which was dependent on the order of execution. Now it's relative to their sorting order values.
- Finally, add the `Undo` and `Redo` items.

### CODE REVIEW

**Comment by MichiGandi** [comment]
I am not sure if the `#`-Character is the best option to separate item-name and sorting order.
Paths would look like this: `File#1/Save#5`

Please some Feedback.

File: Assets/Scripts/System/ContextMenuSystem.cs [hunk 1]
```diff
@@ -55,6 +73,44 @@ public void OpenSubmenu(Guid menuId, List<ContextMenuState.Placement> possiblePl
             _editorEvents.InvokeUpdateContextMenuEvent();
         }
 
+        /// <summary>
+        /// Open the given context menu and sort all its items by their sorting priotiry.
+        /// </summary>
+        /// <param name="position">The position used as origin for the context menu.</param>
+        /// <param name="items">All the menu items for the context menu.</param>
+        public void OpenMenuSorted(Vector3 position, List<ContextMenuItem> items)
```
Comment (cblech) [inline Assets/Scripts/System/ContextMenuSystem.cs]: This is never used. And it calles it self on every path, resulting in an infinite loop when called

File: Assets/Scripts/System/ContextMenuSystem.cs [hunk 2]
```diff
@@ -55,6 +73,44 @@ public void OpenSubmenu(Guid menuId, List<ContextMenuState.Placement> possiblePl
...
+
+            OpenMenuSorted(position, items);
+        }
+
+        /// <summary>
+        /// Open the given context menu and sort all its items by their sorting priotiry.
+        /// </summary>
+        /// <param name="possiblePlacements">The Placement options for the context menu.</param>
+        /// <param name="items">All the menu items for the context menu.</param>
+        public void OpenMenuSorted(List<ContextMenuState.Placement> possiblePlacements, List<ContextMenuItem> items)
```
Comment (cblech) [inline Assets/Scripts/System/ContextMenuSystem.cs]: Also, never used

File: Assets/Scripts/System/ContextMenuSystem.cs [hunk 3]
```diff
@@ -55,6 +73,44 @@ public void OpenSubmenu(Guid menuId, List<ContextMenuState.Placement> possiblePl
...
+        }
+
+        //TODO: specify: <param name="menuId"></param>
+        /// <summary>
+        /// Open the given context menu and sort all its items by their sorting priotiry.
+        /// </summary>
+        /// <param name="menuId"></param>
+        /// <param name="possiblePlacements">The Placement options for the context menu.</param>
+        /// <param name="items">All the menu items for the context menu.</param>
+        public void OpenSubmenuSorted(Guid menuId, List<ContextMenuState.Placement> possiblePlacements, List<ContextMenuItem> items)
```
Comment (cblech) [inline Assets/Scripts/System/ContextMenuSystem.cs]: Also, never used and infinite loop

**Review by cblech — CHANGES_REQUESTED** [review]

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m62y51 Add "Edit → Undo"/"Edit → Redo" buttons to Menu Bar](https://app.clickup.com/t/861m62y51)

File: Assets/Scripts/System/ContextMenuSystem.cs (hunk 1 above)
Comment (MichiGandi) [reply Assets/Scripts/System/ContextMenuSystem.cs]: Infinite recursion is removed.

File: Assets/Scripts/System/ContextMenuSystem.cs (hunk 3 above)
Comment (MichiGandi) [reply Assets/Scripts/System/ContextMenuSystem.cs]: Infinite recursion is removed.

**Review by tcdrop — APPROVED** [review]

**Review by cblech — APPROVED** [review]

**Pull request merged**